### PR TITLE
tools: add linux thermal monitoring and testing tool tmon

### PR DESCRIPTION
**Pull request by feckert**
A Monitoring and Testing Tool for Linux kernel thermal subsystem. This could be helpfull for new targets to test this subsystem.

### CODE REVIEW

**Comment by raenye** [comment]
Why under "_Development_"? maybe "_Global build settings->Kernel build options_" is more appropriate?

**Comment by PolynomialDivision** [comment]
Why not putting this to openwrt/packages?

**Comment by chunkeey** [comment]
Does this work with SDK builds? 

I vividly remembering scratching my head while trying to fix the "spidev-test" utility (which is also part of the kernel source) . Ultimately the third attempt b479db9062b721776be44b976961a1031c1344ea made the build-errors go away and so far it seems to hold. From what I remember, you would just need to add `tools/thermal/tmon` path to `USERSPACE_UTILS_FILES`.

**Comment by feckert** [comment]
> Why under "_Development_"? maybe "_Global build settings->Kernel build options_" is more appropriate?

@raenye  This is not kernel stuff. This is a program to check and evaluate the thermal subsystem of the kernel during development.

> Why not putting this to openwrt/packages?

@PolynomialDivision I can do that too. I just thought that it might be better off in core, since it is built closely with the kernel sources.

> Does this work with SDK builds?

@chunkeey I have not tried that yet. But I don't see why. Or are the kernel sources not available?

**Comment by robimarko** [comment]
For SDK, something like this is needed so the sources get included:
https://github.com/openwrt/openwrt/commit/eecb582a0f85ce8ae2d9525ba3fcaba289df9476

File: package/devel/tmon/Makefile [hunk 1]
```diff
@@ -0,0 +1,58 @@
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=tmon
+PKG_VERSION:=$(shell basename $(LINUX_DIR) | sed 's/linux-//')
```
Comment (hauke) [inline package/devel/tmon/Makefile:5]: perf and spidev_test are using:
````
PKG_RELEASE:=$(LINUX_VERSION)
````
Why do you do it different here?

Comment (feckert) [reply package/devel/tmon/Makefile:5]: Thanks for the hint, I'll take another look at that.

**Comment by hauke** [comment]
Please refresh the patches
> Run . .github/workflows/scripts/ci_helpers.sh
> target/linux/generic/pending-5.15/980-tools-thermal-tmon-fix-compilation-warning-for-wrong.patch
> target/linux/generic/pending-5.15/981-tools-thermal-tmon-fix-compilation-error-for-mips-ta.patch
> Kernel patches for bmips/bcm6318 require refresh. (run 'make target/linux/refresh' and force push this pr)
> You can also check the provided artifacts with the refreshed patch from this CI run.

Please send the patches also to upstream Linux kernel.

**Comment by feckert** [comment]
@hauke refreshed the patche for 5.15 and 6.1. Also send one of the patch [upstream](https://lore.kernel.org/all/20231204141335.2798194-1-fe@dev.tdt.de/) the other is was not necessary.

**Pull request merged**